packages/gatsby/src/commands/serve.js

```diff
@@ -85,9 +85,18 @@ module.exports = async program => {
   app.use(telemetry.expressMiddleware(`SERVE`))
 
   router.use(compression())
-  router.use(express.static(`public`))
+  router.use(express.static(`public`, { redirect: false }))
   const matchPaths = await readMatchPaths(program)
   router.use(matchPathRouter(matchPaths, { root }))
+  router.use((req, res, next) => {
+    if (!/.+\..+$/g.test(req.url) && req.url.slice(-1) !== `/`) {
```

Comment:
for clarity, could we store this regex in a variable that has a descriptive name? mentally grokking this regex is a bit confounding 😆

Reply:
or at least add a comment above explaining what this is doing, what it's solving, etc. I'm just worried about the long term maintenance of this fix getting lost in 6-12m

Reply:
this is copied from my examples, and this is probably not good test to have in retrospect (it infer that resource should be file if it has `.` in it, but that's true at all), so we should remove at least first part of condition leaving only test for trailing slash.

I wonder what we will get now for legitimate 404s tho, now that `serve-static` won't immediately return 404